packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -1021,19 +1023,35 @@ export class IntervalCollectionIterator<TInterval extends ISerializableInterval>
 
 export interface IIntervalCollectionEvent<TInterval extends ISerializableInterval> extends IEvent {
     /**
-     * This event is invoked whenever the properties or endpoints of an interval may have changed.
+     * This event is invoked whenever the endpoints of an interval may have changed.
      * This can happen on:
-     * - endpoint modification (local or remote)
+     * - local endpoint modification
      * - ack of an endpoint modification
-     * - property change (local or remote)
      * - position change due to segment sliding (will always appear as a local change)
      * The `interval` argument reflects the new values.
+     * `previousEndpoints` contains `ReferencePosition`s at the same location as the interval's original endpoints.
+     * These references may be transient, so should only be used for location information.
+     * `local` reflects whether the change originated locally.
+     * `op` is defined if and only if the server has acked this change.
      */
     (event: "changeInterval",
-        listener: (interval: TInterval, local: boolean, op: ISequencedDocumentMessage | undefined) => void);
+        listener: (interval: TInterval, previousEndpoints: { start: ReferencePosition, end: ReferencePosition }, local: boolean, op: ISequencedDocumentMessage | undefined) => void);
```

Comment:
adding in the previous endpoints is a bit tricky here: for ops which explicitly change one or more points of the interval, we will just have the previous local references in hand at the time we emit the event. For sliding it's a bit harder: we'll need to plumb some previous information through some of the callbacks we have set up and then create a transient reference at the same segoff to pass to the user.